src/transformers.ts

```diff
@@ -0,0 +1,291 @@
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/**
+ * Available transformer functions for processing JSON output
+ */
+
+export type TransformerFunction = (data: any) => any;
+
+/**
+ * Flattens nested section arrays into a single array
+ */
+export function flattenTransformer(data: any): any {
+  if (!data || !data.content || !Array.isArray(data.content)) {
+    return data;
+  }
+
+  const flattenedContent = data.content.map((section: any) => {
+    if (section.section && Array.isArray(section.section)) {
+      return {
+        ...section,
+        section: section.section.flat()
+      };
+    }
+    return section;
+  });
+
+  return {
+    ...data,
+    content: flattenedContent
+  };
+}
+
+/**
+ * Removes metadata from all sections
+ */
+export function stripMetadataTransformer(data: any): any {
+  if (!data || !data.content || !Array.isArray(data.content)) {
+    return data;
+  }
+
+  const strippedContent = data.content.map((section: any) => ({
+    section: section.section
+  }));
+
+  return {
+    ...data,
+    content: strippedContent
+  };
+}
+
+/**
+ * Compacts the output by removing empty sections and null values
+ */
+export function compactTransformer(data: any): any {
+  if (!data) return data;
+
+  const compactObject = (obj: any): any => {
+    if (Array.isArray(obj)) {
+      return obj
+        .map(compactObject)
+        .filter(item => item !== null && item !== undefined);
+    }
+    
+    if (obj && typeof obj === 'object') {
+      const compacted: any = {};
+      for (const [key, value] of Object.entries(obj)) {
+        const compactedValue = compactObject(value);
+        if (compactedValue !== null && compactedValue !== undefined) {
+          if (Array.isArray(compactedValue) && compactedValue.length === 0) {
+            continue; // Skip empty arrays
+          }
+          if (typeof compactedValue === 'object' && Object.keys(compactedValue).length === 0) {
+            continue; // Skip empty objects
+          }
+          compacted[key] = compactedValue;
+        }
+      }
+      return compacted;
+    }
+    
+    return obj;
+  };
+
+  return compactObject(data);
+}
+
+/**
+ * Helper function to find and transform properties with '/content/dam/' values
+ */
+function findAndTransformDamReferences(obj: any, references: any[] = []): any {
+  if (typeof obj === 'string' && obj.includes('/content/dam/')) {
+    // Transform the string to the required object format
+    const transformedValue = {
+      _publishUrl: obj.startsWith('/content/dam/') ? `https://odin.adobe.com${obj}` : obj
```

Comment:
Lets add a todo here to not forget to make this more generic and not hard code the odin host.